libsplinter/src/service/scabbard/client/scar.rs

```diff
@@ -0,0 +1,372 @@
+// Copyright 2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::env::{split_paths, var_os};
+use std::fs::File;
+use std::io::Read;
+use std::path::{Path, PathBuf};
+
+use bzip2::read::BzDecoder;
+use tar::Archive;
+
+use super::Error;
+
+const SCAR_FILE_EXTENSION: &str = "scar";
+const SCAR_PATH_ENV_VAR: &str = "SCAR_PATH";
+const MANIFEST_FILENAME: &str = "manifest.yaml";
+const WASM_FILE_EXTENSION: &str = "wasm";
+
+/// The definition of a WASM smart contract, including the bytes of the smart contract itself and
+/// the associated metadata that is required for submitting the smart contract to scabbard.
+#[derive(Debug)]
+pub struct SmartContract {
```

Comment:
This needs to be renamed because we are using this for a high-profile trait in transact.

Reply:
Renamed structs to `SabreSmartContractDefinition` and `SabreSmartContractMetadata`.